CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -5,33 +5,48 @@
 #include <algorithm>
 
 #include "CUDADataFormats/Track/interface/TrajectoryStateSoAT.h"
-#include "Geometry/CommonTopologies/interface/SimplePixelTopology.h"
 #include "HeterogeneousCore/CUDAUtilities/interface/HistoContainer.h"
-
+#include "Geometry/CommonTopologies/interface/SimplePixelTopology.h"
 #include "CUDADataFormats/Common/interface/HeterogeneousSoA.h"
+#include "DataFormats/Common/interface/CMS_CLASS_VERSION.h"
 
 namespace pixelTrack {
+
   enum class Quality : uint8_t { bad = 0, edup, dup, loose, strict, tight, highPurity, notQuality };
   constexpr uint32_t qualitySize{uint8_t(Quality::notQuality)};
   const std::string qualityName[qualitySize]{"bad", "edup", "dup", "loose", "strict", "tight", "highPurity"};
   inline Quality qualityByName(std::string const &name) {
     auto qp = std::find(qualityName, qualityName + qualitySize, name) - qualityName;
     return static_cast<Quality>(qp);
   }
+
 }  // namespace pixelTrack
 
+// This definition below seems to be necessary to ROOT to avoid DictionaryNotFound error for old TrackingRecHit2DHeterogeneous
+// in RAW samples in which the Run3 HLT has already been run. Without this those samples are useless beacuse of the abovementioned
+// ROOT dictionary error. Alternative solution would be to impose a drop of HLT collections in RECO step, but seems not reasonable.
+// Same reasoning applies to TrackingRecHit2DHeterogeneous.
```

Comment:
Could you clarify whether you mean RAW _data_ samples, or DIGI-RAW MC samples ?
AFAIK the RAW _data_ files should not contain any of these collections, and so they should not cause any issues.

Reply:
Indeed I meant DIGI-RAW MC.

Reply:
Ah, OK.

Is that something we care much about ?
Would breaking this kind of backwards compatibility simplify the PR significantly ?

Reply:
The problem is not about a product in the input file, but a product anywhere in the processing history of any product in the input file (that is not explicitly dropped on input). I suspect we'd hit the same problem with the Run 3 RAW data as well since the `TrackSoAHeterogeneousT` is used in the HLT reconstruction.

The framework's current requirement on the job to have a dictionary for data products that are part of the processing history but not present in the input file will be removed in #38806. Once that gets merged, this PR can drop the `TrackSoAHeterogeneousT`.

Reply:
> The framework's current requirement on the job to have a dictionary for data products that are part of the processing history but not present in the input file will be removed in #38806. Once that gets merged, this PR can drop the `TrackSoAHeterogeneousT`.

Just to note that #38806 was merged for the next IB.

Reply:
Thanks Matti, let me update this accordingly.